### DISCUSSION TITLE
NodeGraph add support for SQLite

### DISCUSSION THREAD

**Discussion by nickper** [grafana/grafana#48629] [Data Sources]
I want to have to use my SQLite database to be used for NodeGraph. Currently there is only X-ray known to be supported, but I want to use a local database instead.

**Comment by connorlindsey** [comment]
@nickper we use the NodeGraph visualization for Tempo, Jaeger, and Zipkin as well. We also recently made it a bit more flexible to work with any datasource. See the docs for instructions on how to use it: https://grafana.com/docs/grafana/latest/visualizations/node-graph/#data-api

You'll likely want to run one queries for nodes and one for edges from the panel you're using it in.

**Reply by yosiasz** [reply]
Greetings,

Doing this for mysql data source, one for nodes and another for edges but still hitting a wall. Am I missing something in my setup?

![image](https://user-images.githubusercontent.com/900951/170745413-c33fa0d5-fada-4546-aa61-ede9b9243bad.png)

**Reply by connorlindsey** [reply]
Does it render properly if you toggle the Table view?

**Reply by yosiasz** [reply]
Thanks @nickper it works fine now after upgrade to 8.5.2.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
